### ISSUE TITLE
False positive "Cannot move" inspection on static variable that has Copy trait

### ISSUE THREAD

**Issue by aloucks** [intellij-rust/intellij-rust#5165]
## Environment

* **IntelliJ Rust plugin version:** Rust (0.2.118.2171-193)
* **Rust toolchain version:** rustc 1.42.0 (b8cedc004 2020-03-09)
* **IDE name and version:**  IntelliJ IDEA 2019.3.4 Build #IC-193.6911.18 March 17, 2020
* **Operating system:** Windows 10 amd64

I've noticed this with the `glfw` crate's global `FAIL_ON_ERRORS` static variable.

![image](https://user-images.githubusercontent.com/221559/77493663-180e0b00-6e1a-11ea-809d-8c6e12f7894d.png)

The type is `Copy`.

https://github.com/PistonDevelopers/glfw-rs/blob/c51524b52e2e5cd7411b0e7d1cfe3e4eb110d4d6/src/lib.rs#L428

https://github.com/PistonDevelopers/glfw-rs/blob/c51524b52e2e5cd7411b0e7d1cfe3e4eb110d4d6/src/lib.rs#L377

EDIT: This is in the most recent IntelliJ Rust as of 2020-03-34 on Windows 10.

_Originally posted by @aloucks in https://github.com/intellij-rust/intellij-rust/issues/4850#issuecomment-603596901_

**Comment by ambye85** [comment]
![Screenshot 2020-05-18 at 10 34 10](https://user-images.githubusercontent.com/17502031/82197705-36145b00-98f3-11ea-8de9-d1ea6524da70.png)

Same issue with plugin version 0.2.121.3059-201-stable on:

CLion 2020.1.1
Build #CL-201.7223.86, built on April 29, 2020
Runtime version: 11.0.6+8-b765.40 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.4

**Comment by codedcosmos** [comment]
Still an issue with the rust plugin from JetBrains 0.3.139.3615-203

IntelliJ IDEA 2020.3 (Community Edition)
Build #IC-203.5981.155, built on December 1, 2020
Runtime version: 11.0.9+11-b1145.21 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.8.0-43-generic
GC: ParNew, ConcurrentMarkSweep
Non-Bundled Plugins: GLSL, org.toml.lang, org.rust.lang

**Comment by ortem** [comment]
Thank you for the reports!

I've finally started working on this and found the minimal reproducible example:
```Rust
#[derive(Copy, Clone)]
struct Bar<Fn> { a: Fn }

fn bar(x: Bar<fn()>) {
    x;
    x; // false-positive move error
}
```

Although the following case without type parameter works just fine:
```Rust
#[derive(Copy, Clone)]
pub struct Foo { a: fn() }

fn foo(x: Foo) {
    x;
    x; // OK
}
```

<img width="429" alt="Screenshot 2021-02-11 at 14 38 41" src="https://user-images.githubusercontent.com/4854600/107631961-e3e04a80-6c76-11eb-915f-efa3502ace01.png">

So I'll try to investigate and probably fix the problem soon.
